common/schema-builder/src/main/java/io/smallrye/graphql/schema/creator/type/TypeCreator.java

```diff
@@ -45,24 +40,28 @@ public TypeCreator(ReferenceCreator referenceCreator, FieldCreator fieldCreator,
     @Override
     protected void addFields(Type type, ClassInfo classInfo, Reference reference) {
         // Fields
-        List<MethodInfo> allMethods = new ArrayList<>();
+        Map<String, MethodInfo> allMethods = new HashMap<>();
         Map<String, FieldInfo> allFields = new HashMap<>();
 
         // Find all methods and properties up the tree
         for (ClassInfo c = classInfo; c != null; c = ScanningContext.getIndex().getClassByName(c.superName())) {
             if (InterfaceCreator.canAddInterfaceIntoScheme(c.toString())) { // Not java objects
                 List<MethodInfo> classMethods = filterOutBridgeMethod(c.methods());
-                allMethods.addAll(classMethods);
-                allMethods.addAll(getAllInterfaceMethods(c, classMethods
+                for (MethodInfo classMethod : classMethods) {
+                    allMethods.putIfAbsent(classMethod.name(), classMethod);
```

Comment:
Hold on, why exactly do you need to turn the List into a Map and use `classMethod.name()` as a key? That will clash in case that the method has overrides, no?

Reply:
From my understanding, GraphQL doesn't support overloading (both param and return type), i.e. in each GraphQL Object, there cannot be two field with the same name even if they have different param types. Therefore, I think having `MethodInfo.toString()` can be replaced by `MethodInfo.name()`.

In the case of Java methods with the same name, the current implementation puts them both in the `allMethods` list. Then in the following loop
```
        for (MethodInfo methodInfo : allMethods.values()) {
            if (MethodHelper.isPropertyMethod(Direction.OUT, methodInfo)) {
                String fieldName = MethodHelper.getPropertyName(Direction.OUT, methodInfo.name());
                FieldInfo fieldInfo = allFields.remove(fieldName);
                fieldCreator.createFieldForPojo(Direction.OUT, fieldInfo, methodInfo, reference).ifPresent(type::addField);
            }
        }
```
Both of the two Java methods get a call to `type::addField` whose implementation is basically `LinkedHashMap::put` with key `fieldName`, therefore, the second method in the list will override the previous one implicitly.

My change from `List<MethodInfo>` to `Map<String, MethodInfo>` aims to bring this implicit overwrite explicit by maintaining one method per fieldName when we are looping over the Java Inheritance Tree.

My intentioned code behavioural change is
1. the same order for Java Inheritance Tree walk,
2. for methods with the same fieldName, instead of the last encounter overwrites, the earliest encounter takes priority.

Reply:
GraphQL doesn't have overloading, but Java does, and here we're talking about the name of the Java method, not of the resulting GraphQL field. We can have multiple methods with the same name, differentiated using a `@Name` annotation to denote the name of the GraphQL field to be created from them

Reply:
There's some failures in the tests, please try to run the test suite, it's probably related

Reply:
> GraphQL doesn't have overloading, but Java does, and here we're talking about the name of the Java method, not of the resulting GraphQL field. We can have multiple methods with the same name, differentiated using a `@Name` annotation to denote the name of the GraphQL field to be created from them

Indeed I didn't know about `@Name` directive. Now that makes sense to me. Then another possible fix is that instead of `List<MethodInfo> allMethods`, we put them in different categories depending how many layers between the method to the class, i.e. `List<List<MethodInfo>> allMethods`. And then we process the layers furtherest according to my logic above.

I will add a commit for it.

> There's some failures in the tests, please try to run the test suite, it's probably related

The failures are because I added the methods according to the order `for (MethodInfo methodInfo : allMethods.values())`, which is not determined order for HashMap. And the test relies on a specific order of adding fields.

Additional Note, very unfortunate we need to do one message per day, I live in the EST zone btw.

Reply:
Cool, it looks good to me now, I'll open more conversations if I find more issues.
I am in CET and unfortunately usually finish work quite early, but I can try to have a look in the evening too, hopefully